src/matrix.js

```diff
@@ -802,19 +802,14 @@ export class AbstractMatrix {
     return diag;
   }
 
-  norm(type = 'frobenius') {
-    let result = 0;
-    if (type === 'max') {
-      return this.max();
-    } else if (type === 'frobenius') {
-      for (let i = 0; i < this.rows; i++) {
-        for (let j = 0; j < this.columns; j++) {
-          result = result + this.get(i, j) * this.get(i, j);
-        }
-      }
-      return Math.sqrt(result);
-    } else {
-      throw new RangeError(`unknown norm type: ${type}`);
+  norm(type) {
+    switch (type ?? 'frobenius') {
```

Comment:
Please keep the default parameter as it's more idiomatic JS

Reply:
My rationale here was that this would allow for `norm(undefined)`. TIL that default parameters are used when the argument evaluates to `undefined`, not just when it's omitted!